drivers/sensor/lsm6dsl/lsm6dsl_spi.c

```diff
@@ -0,0 +1,106 @@
+/* lsm6dsl_spi.c - SPI routines for LSM6DSL driver
+ */
+
+/*
+ * Copyright (c) 2018 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <spi.h>
+#include "lsm6dsl.h"
+
+#define LSM6DSL_SPI_READ		(1 << 7)
+
+#define SPI_CS NULL
+
+static struct spi_config lsm6dsl_spi_conf = {
+	.frequency = 16000000,
```

Comment:
you'll need to make it Kconfig based, because if the spi master controller device is not able to go that high (it's not high, but some controllers are twice slower than that) is won't work on these controller.

Reply:
ack